src/reports_api/resources/project.py

```diff
@@ -0,0 +1,58 @@
+"""Resource for project endpoints."""
+from http import HTTPStatus
+from flask import current_app
+
+from flask_restx import Namespace, Resource, cors
+
+from reports_api.services import ProjectService
+from reports_api.utils.util import cors_preflight
+
+from reports_api.models import Project
+
+
+API = Namespace('projects', description='Projects')
+
+
+@cors_preflight('GET')
+@API.route('', methods=['GET', 'POST', 'OPTIONS'])
+class Projects(Resource):
+    """Endpoint resource to manage projects."""
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def get():
+        """Return all projects."""
+        return ProjectService.find_all(), HTTPStatus.OK
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def post():
+        """Create new project"""
+        project = ProjectService.create_project(API.payload)
+        return project.as_dict(), HTTPStatus.CREATED
+
+
+@cors_preflight('GET')
+@API.route('/<int:project_id>', methods=['GET', 'PATCH', 'DELETE', 'OPTIONS'])
```

Comment:
Is it "PATCH" really. This is a full update.right?

Reply:
The main use of an update API I found was marking project completed, when withdrawn or terminated or completed, that's why opted for Patch

Reply:
Updated